game-core/src/main/java/games/strategy/triplea/ai/pro/util/ProPurchaseValidationUtils.java

```diff
@@ -0,0 +1,279 @@
+package games.strategy.triplea.ai.pro.util;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.Constants;
+import games.strategy.triplea.Properties;
+import games.strategy.triplea.ai.pro.ProData;
+import games.strategy.triplea.ai.pro.data.ProPlaceTerritory;
+import games.strategy.triplea.ai.pro.data.ProPurchaseOption;
+import games.strategy.triplea.ai.pro.data.ProPurchaseTerritory;
+import games.strategy.triplea.ai.pro.data.ProResourceTracker;
+import games.strategy.triplea.ai.pro.simulate.ProDummyDelegateBridge;
+import games.strategy.triplea.attachments.TerritoryAttachment;
+import games.strategy.triplea.delegate.AbstractPlaceDelegate;
+import games.strategy.triplea.delegate.Matches;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import org.triplea.java.collections.CollectionUtils;
+
+/** Pro AI purchase utilities. */
```

Comment:
Is this comment here because it has to be? I wonder as `ValidationUtils` seems to be more specific than the comment implies. Second, have you considered using `@UtilityClass` to avoid the private constructor?

Reply:
Updated comment. Added that annotation to this class and ProPurchaseUtils.